test/e2e/proxy_test.go

```diff
@@ -289,6 +289,50 @@ func TestProxyFlow(t *testing.T) {
 				_, err := hostAwaitWithShorterTimeout.CreateAPIProxyClient(t, user.token, proxyWorkspaceURL)
 				require.EqualError(t, err, `an error on the server ("unable to get target cluster: the requested space is not available") has prevented the request from succeeding`)
 			})
+
+			t.Run("invalid request headers", func(t *testing.T) {
+				// given
+				proxyWorkspaceURL := hostAwait.ProxyURLWithWorkspaceContext(user.compliantUsername)
+				rejectedHeaders := []headerKeyValue{
+					{"Impersonate-Group", "system:cluster-admins"},
+					{"Impersonate-Group", "system:node-admins"},
+				}
+				client := http.Client{
+					Timeout: time.Duration(5 * time.Second), // because sometimes the network connection may be a bit slow
+				}
+				client.Transport = &http.Transport{
+					TLSClientConfig: &tls.Config{
+						InsecureSkipVerify: true, // nolint:gosec
+					},
+				}
+				t.Logf("proxyWorkspaceURL: %s", proxyWorkspaceURL)
+				nodesURL := fmt.Sprintf("%s/api/v1/nodes", proxyWorkspaceURL)
+				t.Logf("nodesURL: %s", nodesURL)
+
+				for _, header := range rejectedHeaders {
+					t.Run(fmt.Sprintf("k=%s,v=%s", header.key, header.value), func(t *testing.T) {
+						// given
+						request, err := http.NewRequest("GET", nodesURL, nil)
+						request.Header.Add(header.key, header.value)
+						require.NoError(t, err)
+						request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", user.token)) // uses the user's token with the impersonation headers
+
+						// when
+						resp, err := client.Do(request)
+
+						// then
+						require.NoError(t, err)
+						require.NotNil(t, resp)
+						require.Equal(t, 403, resp.StatusCode)
+						r, _ := io.ReadAll(resp.Body)
+						defer func() {
+							resp.Body.Close()
+						}()
+						assert.NotContains(t, string(r), "NodeList")
```

Comment:
maybe we could add a comment here explaining why this should not contain that `NodeList` or we could check for an empty response body ? I don't actually recall what is returned as body when the `StatusCode` is `403` ...

Reply:
The response is 
```
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"nodes is forbidden: User \"proxymember1\" cannot list resource \"nodes\" in API group \"\" at the cluster scope","reason":"Forbidden","details":{"kind":"nodes"},"code":403}
```
Checking status code is 403 is probably good enough but I changed the assertion to check for the specific error message just to be more thorough. See 6128d0f356b76102a47fa2e9034a1fa269af05a2

Reply:
Thanks, this makes it more clear, at least for me!